test/prepare.js

```diff
@@ -30,3 +46,66 @@ test('compiling with several and statements', t => {
     t.end()
   }
 })
+
+test('prepare query with projection, sourceProjection, and geometry, then execute filter on non-4326 source data', (t) => {
+  t.plan(3)
+  const options = {
+    geometry: {
+      xmin: -118.15738230943678,
+      ymin: 34.179713563470166,
+      xmax: -118.15718114376067,
+      ymax: 34.18019950919287,
+      spatialReference: {
+        wkid: 4326
+      }
+    },
+    sourceDataSR: caStatePlaneWKT
+  }
+  const query = Winnow.prepareQuery(options)
+  const filtered = query(trees)
+  t.equal(filtered.features.length, 3)
+  t.equal(filtered.features[0].geometry.coordinates[0], 6514083.848741564)
+  t.equal(filtered.features[0].geometry.coordinates[1], 1887939.4934484742)
+})
+
+test('prepare query with projection, sourceProjection, and geometry, then execute non-4326 filter on non-4326 source data', (t) => {
+  t.plan(3)
+  const options = {
+    geometry: {
+      xmin: 6514066.3001615712419152,
+      ymin: 1887820.5006760144606233,
+      xmax: 6514127.4193187793716788,
+      ymax: 1887997.4401315276045352,
+      spatialReference: {
+        wkt: caStatePlaneWKT
+      }
+    },
+    sourceDataSR: caStatePlaneWKT
+  }
+  const query = Winnow.prepareQuery(options)
+  const filtered = query(trees)
+  t.equal(filtered.features.length, 3)
+  t.equal(filtered.features[0].geometry.coordinates[0], 6514083.848741564)
+  t.equal(filtered.features[0].geometry.coordinates[1], 1887939.4934484742)
+})
+
+test('query with projection, sourceProjection, and geometry, then execute filter on non-4326 source data', (t) => {
+  t.plan(3)
+  const options = {
+    geometry: {
+      xmin: -118.15738230943678,
+      ymin: 34.179713563470166,
+      xmax: -118.15718114376067,
+      ymax: 34.18019950919287,
+      spatialReference: {
+        wkid: 4326
+      }
+    },
+    sourceDataSR: caStatePlaneWKT
```

Comment:
can we make this `sourceSR` to be more consistent with `inSR` and `outSR`

Reply:
also can we support setting a crs on geojson instead of `sourceSR`?

Reply:
I did consider inspecting the `crs` property on the geojson.  The [`prepareQuery`](https://github.com/koopjs/winnow/blob/master/src/index.js#L32-L34) function generates the geometry envelop and adds it to the parameter array before the source data is available - so it doesn't have access to a geojson `crs` property for reprojecting the geometry envelope.  

`prepareQuery` could however be altered so that the geometry parameter was generated when the input data was received (and thus exposing `crs` from data) - but this could lead to inefficiencies.   For example, in opendata-koop, we would end up reprojecting the envelope for ever feature in a dataset, since they are streamed one by one through the query execution function.  You could perhaps avoid this by getting the `crs` from the first feature of the set, using it to create the geometry parameter, and then not repeating that operation again.  It seemed like a more complicated approach, and since `crs` has been removed from GeoJSON spec, I didn't go with it.  Happy to reconsider based on your reaction to the above.

Reply:
Let's stick with what you've got already.

Reply:
Changed to `sourceSR` in 637f6cc.